lib/dialects/abstract/query-generator.js

```diff
@@ -478,6 +478,11 @@ module.exports = (function() {
         options.using = options.indexType || options.method;
       }
 
+      // table name can be either a string or an object
+      if (Utils._.isObject(tableName)) {
+        tableName = tableName.schema + '.' + tableName.tableName;
```

Comment:
Actually this code could be removed entirely, just change `quoteIdentifiers` to `quoteTable` in the bottom.